### PR TITLE
Update cache files v3

### PR DESCRIPTION
**Pull request by sawer297**
added Http.Cache.expires.time parameter to bes config files and code that uses same in RemoteResources

### CODE REVIEW

File: http/HttpCache.cc [hunk 1]
```diff
@@ -85,6 +85,22 @@ namespace http {
         return size_in_megabytes;
     }
 
+    unsigned long long HttpCache::getCacheExpiresTime() {
```
Comment (jgallagher59701) [inline http/HttpCache.cc:88]: Change this to return 'unsigned long' to match the type of 'time_in_seconds'.

File: http/HttpCache.h [hunk 2]
```diff
@@ -97,6 +97,7 @@ namespace http {
         virtual std::string get_cache_file_name(const std::string &src, bool mangle=true);
 #endif
 
+        static unsigned long long getCacheExpiresTime();
```
Comment (jgallagher59701) [inline http/HttpCache.h:100]: unsigned long long --> unsigned long here too. Also, I'm not sure I understand why this method is 'static'.

File: http/HttpCache.cc [hunk 3]
```diff
@@ -85,6 +85,22 @@ namespace http {
...
+        bool found = false;
+        string time;
+        unsigned long time_in_seconds = 0;
+        TheBESKeys::TheKeys()->get_value(HTTP_CACHE_EXPIRES_TIME_KEY, time, found);
+
+        if (found) {
+            std::istringstream iss(time);
+            iss >> time_in_seconds;
+        } else {
+            time_in_seconds = 3600;
```
Comment (jgallagher59701) [inline http/HttpCache.cc:98]: Use a constant like 'REMOTE_RESOURCE_DEFAULT_EXPIRED_INTERVAL' here instead of 3600.

File: http/RemoteResource.h [hunk 4]
```diff
@@ -40,8 +40,9 @@
 
 #include "RemoteResource.h"
 #include "rapidjson/document.h"
+#include "HttpCache.h"
 
-#define REMOTE_RESOURCE_DEFAULT_EXPIRED_INTERVAL 3600
+//#define REMOTE_RESOURCE_DEFAULT_EXPIRED_INTERVAL -1
```
Comment (jgallagher59701) [inline http/RemoteResource.h:45]: put a #define for REMOTE_RESOURCE_DEFAULT_EXPIRED_INTERVAL in HttpNames.h

**Review by jgallagher59701 — CHANGES_REQUESTED** [review]

**Review by ndp-opendap — CHANGES_REQUESTED** [review]
James' suggestions are good, then it's done.

File: http/HttpCache.h (hunk 2 above)
Comment (sawer297) [reply http/HttpCache.h:100]: its static because i just copy/pasted the other method that retrieves the size and changed it to suit

**Pull request merged**